packages/webpack-graph-ql/src/lib/types/auto-schema.ts

```diff
@@ -0,0 +1,133 @@
+
+export interface AutoSchemaCons {
+    getFields(): Map<string, any>;
+    getQueries(): Map<string, any>;
+}
+
+const typeMap = new Map();
+const queryMap = new Map();
+const mutationMap = new Map();
+const fieldMap = new Map();
+
+/**
+ * The `AutoSchema` class is the base class for GraphQL types. It
+ * provides a set of static methods that can be used to define and retrieve
+ * the fields and queries of a GraphQL type.
+ *
+ * By using the `AutoSchema` class as a base class for their GraphQL types and
+ * defining the fields and queries of their types using the `getFields` and
+ * `getQueries` methods, developers can automatically generate a GraphQL
+ * schema for their types. This can be useful for generating a schema on the
+ * fly or for dynamically modifying a schema based on the data being queried.
+ */
+export abstract class AutoSchema {
+
+  private static _type: any = null;
+  private static _arrayType: any = null;
+
+  /**
+   * Returns the GraphQLObjectType object for this class.
+   *
+   * The object type represents a group of fields in the GraphQL schema and is
+   * created using lazy initialization, meaning it is only created when it is
+   * first needed.
+   *
+   * @return {GraphQLObjectType} The GraphQLObjectType object for this class.
+   */
+  static get type() {
+    if (!this._type) {
+      const fields = 
+      this._type = new GraphQLObjectType({
+        name: this.name,
+        fields: this.getFieldConfigs()
+      });
+    }
+    return this._type;
+  }
+
+  /**
+   * Returns the GraphQLList object for this class.
+   *
+   * The GraphQLList object represents an array of objects in the GraphQL schema
+   * and is created using lazy initialization, meaning it is only created when it
+   * is first needed.
+   *
+   * @return {GraphQLList} The GraphQLList object for this class.
+   */
+  static get arrayType() {
+    if (!this._arrayType) {
+      this._arrayType = new GraphQLList(this.type)
+    }
+    return this._arrayType;
+  }
```

Comment:
Add why to docstring: static methods that return multiple instances like findAssetsByChunkName